source/client/lsp-raw_clients.adb

```diff
@@ -214,19 +214,21 @@ package body LSP.Raw_Clients is
               (1 .. Ada.Streams.Stream_Element_Count (Size))
                 with Import, Address => Slice'Address;
             Last  : Natural;
+
          begin
             Client.Server.Write_Standard_Input
               (Raw, Ada.Streams.Stream_Element_Count (Last));
 
-            if Last = 0 then
```

Comment:
The condition `Last = 0` corresponds to API definition:
```ada
   procedure Write_Standard_Input
     (Self : in out Process'Class;
      Data : Ada.Streams.Stream_Element_Array;
      Last : out Ada.Streams.Stream_Element_Offset);
   --  Do nothing if Self.Status /= Running. If no data was written you will
   --  get Standard_Input_Available notification latter.
```

You can expect next Standard_Input_Available only when `Last < Data'Last`.

Reply:
Behavior of Last parameter and notification has been clarified in source code.